store/src/java/com/zimbra/cs/imap/EhcacheImapCache.java

```diff
@@ -45,7 +45,10 @@
 
     @SuppressWarnings("serial")
     EhcacheImapCache(String name, boolean active) {
-        ehcache = EhcacheManager.getInstance().getEhcache(name);
+        // If running inside mailboxd, share mailboxd cache, else use separate imap cache.  This avoids issues when running
+        // decoupled IMAP service on same host as mailbox.
+        EhcacheManager.Service service = System.getProperty("jetty.home", null) == null ? EhcacheManager.Service.IMAP : EhcacheManager.Service.MAILBOX;
```

Comment:
I think it is better to rely on a property that we can explicitly set on the JVM rather then on an absence of a property.